third_party/java/arcs/build_defs/internal/manifest.bzl

```diff
@@ -37,6 +38,37 @@ def arcs_manifest(name, srcs, deps = [], visibility = None):
         execute = False,
     )
 
+def arcs_serialize_manifest(name, src, deps = [], out = None, visibility = None):
+    """Serialize a manifest file.
+
+    This converts a '.arcs' file into a JSON representation, using manifest2proto.
+
+    Args:
+      name: the name of the target to create
+      src: an Arcs manifest files to serialize
+      deps: list of dependencies (other manifests)
+      visibility: list of visibilities
+    """
+    if not src.endswith(".arcs"):
+        fail("src must be a .arcs file")
+
+    if type(deps) == str:
+        fail("deps must be a list")
+
+    if type(out) != str:
+        fail("out must be a filename (a string), or None")
```

Comment:
These checks might be overkill (and not that comprehensive). They will fail elsewhere if they're wrong, I'd suggest just deleting them. (except the `endswith(".arcs")` one, that's probably useful)